tests/metrics/test_infidelity.py

```diff
@@ -0,0 +1,211 @@
+            n_perturb_samples=5,
+            max_batch_size=1,
+        )
+
+    def test_classification_infidelity_convnet(self):
+        model = BasicModel_ConvNet_One_Conv()
+        dl = DeepLift(model)
+
+        input = torch.stack([torch.arange(16).float()] * 2, dim=0).view(2, 1, 4, 4)
+
+        self.basic_infidelity_helper(
+            model, dl, input, [0.0032, 0.0032], target=0, multi_input=False
+        )
+
+    def test_classification_infidelity_convnet_multi_targets(self):
+        model = BasicModel_ConvNet_One_Conv()
+        ig = IntegratedGradients(model)
+        nt = NoiseTunnel(ig)
+
+        input = torch.stack([torch.arange(16).float()] * 20, dim=0).view(20, 1, 4, 4)
+
+        self.basic_infidelity_helper(
+            model,
+            ig,
+            input,
+            [0.0032] * 20,
+            target=torch.tensor([1] * 20),
+            multi_input=False,
+            n_perturb_samples=500,
+            max_batch_size=10,
+        )
+        self.basic_infidelity_helper(
+            model,
+            nt,
+            input,
+            [0.0001] * 20,
+            target=torch.tensor([1] * 20),
+            multi_input=False,
+            n_perturb_samples=500,
+            max_batch_size=10,
+            nt_type="vargrad",
+        )
+
+    def test_classification_infidelity_tpl_target(self):
+        model = BasicModel_MultiLayer()
+        input = torch.randn(4, 3)
+        additional_forward_args = (None, True)
+        targets = [(0, 1, 1), (0, 1, 1), (1, 1, 1), (0, 1, 1)]
+        sa = Saliency(model)
+
+        infid1 = self.basic_infidelity_helper(
+            model,
+            sa,
+            input,
+            [0.0, 0.0167, 0.0167, 0.0167],
+            additional_args=additional_forward_args,
+            target=targets,
+            multi_input=False,
+        )
+
+        infid2 = self.basic_infidelity_helper(
+            model,
+            sa,
+            input,
+            [0.0, 0.0167, 0.0167, 0.0167],
+            additional_args=additional_forward_args,
+            target=targets,
+            max_batch_size=2,
+            multi_input=False,
+        )
+        assertArraysAlmostEqual(infid1, infid2, 1e-05)
+
+    def basic_infidelity_helper(
+        self,
+        model,
+        attr_algo,
+        inputs,
+        expected,
+        additional_args=None,
+        n_perturb_samples=10,
+        target=None,
+        max_batch_size=None,
+        multi_input=True,
+        **kwargs
+    ):
+        def perturb_func(input1, input2=None):
+            perturb1 = torch.stack(
+                [torch.Tensor([0.0009])] * np.prod(list(input1.shape)), dim=-1
+            ).view(input1.shape)
+            if multi_input:
+                perturb2 = torch.stack(
+                    [torch.Tensor([0.0121])] * np.prod(list(input2.shape)), dim=-1
+                ).view(input2.shape)
+                return (perturb1, perturb2), (input1 + perturb1, input2 + perturb2)
```

Comment:
nit: Based on the signs applied in the infidelity implementation, this maybe should be input1 - perturb1 and input2 - perturb2 or the perturbations should be - perturb1, -perturb2?